clang/runtime/dpct-rt/include/lapack_utils.hpp.inc

```diff
@@ -133,6 +132,260 @@ inline int sygvd(sycl::queue &queue, std::int64_t itype, oneapi::mkl::job jobz,
+    int info_val = static_cast<int>(e.info());
+    queue.memcpy(info, &info_val, sizeof(int)).wait();
+    return 1;
+  } catch (sycl::exception const& e) {
+    std::cerr << "Caught synchronous SYCL exception:" << std::endl
+              << "reason: " << e.what() << std::endl;
+    queue.memset(info, 0, sizeof(int)).wait();
+    return 1;
+  }
+  queue.memset(info, 0, sizeof(int));
+  return 0;
+#endif
+}
+// DPCT_LABEL_END
+// DPCT_LABEL_BEGIN|potrf_batch|dpct::lapack
+// DPCT_DEPENDENCY_BEGIN
+// Util|DataType|UsmRestricted
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+/// Computes the Cholesky factorizations of a batch of symmetric (or Hermitian,
+/// for complex data) positive-definite matrices.
+/// \return Returns 0 if no synchronous exception, otherwise returns 1.
+/// \param [in] queue Device queue where calculations will be performed.
+/// \param [in] uplo Must be uplo::upper or uplo::lower.
+/// \param [in] n The order of the matrix A.
+/// \param [in,out] a Array of pointers to matrix A.
+/// \param [in] lda The leading dimension of matrix A.
+/// \param [out] info If lapack synchronous exception is caught, the value
+/// returned from info() method of the exception is set to \p info.
+/// \param [in] group_size The batch size.
+template <typename T>
+inline int potrf_batch(sycl::queue &queue, oneapi::mkl::uplo uplo, int n,
+                       T *a[], int lda, int *info, int group_size) {
+#ifdef DPCT_USM_LEVEL_NONE
+  throw std::runtime_error("this API is unsupported when USM level is none");
+#else
+  using Ty = typename DataType<T>::T2;
+  struct matrix_info_t {
+    oneapi::mkl::uplo uplo_info;
+    std::int64_t n_info;
+    std::int64_t lda_info;
+    std::int64_t group_size_info;
+  };
+  matrix_info_t *matrix_info =
+      (matrix_info_t *)std::malloc(sizeof(matrix_info_t));
+  matrix_info->uplo_info = uplo;
+  matrix_info->n_info = n;
+  matrix_info->lda_info = lda;
+  matrix_info->group_size_info = group_size;
+  std::int64_t scratchpad_size = 0;
+  sycl::event e;
+  Ty *scratchpad = nullptr;
+  try {
+    scratchpad_size = oneapi::mkl::lapack::potrf_batch_scratchpad_size<Ty>(
+        queue, &(matrix_info->uplo_info), &(matrix_info->n_info),
+        &(matrix_info->lda_info), 1, &(matrix_info->group_size_info));
+    scratchpad = sycl::malloc_device<Ty>(scratchpad_size, queue);
+    e = oneapi::mkl::lapack::potrf_batch(
+        queue, &(matrix_info->uplo_info), &(matrix_info->n_info), (Ty **)a,
+        &(matrix_info->lda_info), 1, &(matrix_info->group_size_info),
+        scratchpad, scratchpad_size);
+  } catch (oneapi::mkl::lapack::exception const &e) {
+    std::cerr << "Unexpected exception caught during call to LAPACK API: "
+                 "potrf_batch_scratchpad_size/potrf_batch"
+              << std::endl
+              << "reason: " << e.what() << std::endl
+              << "info: " << e.info() << std::endl;
+    std::vector<int> info_vec(group_size, static_cast<int>(e.info()));
```

Comment:
Not all of the problems from the batch will get non-zero status info code. There is a `potrf_batch` documentation describing `batch_error` (mistakenly called batch_exceltion in the documentation): https://www.intel.com/content/www/us/en/develop/documentation/oneapi-mkl-dpcpp-developer-reference/top/lapack-routines/potrf_batch-group-version.html

> This exception is thrown when problems occur during calculations. You can obtain the info code of the problem using the info() method of the exception object:
> If info = -n, the n-th parameter had an illegal value.
> If info equals the value passed as scratchpad size, and detail() returns non-zero, then the passed scratchpad is of insufficient size, and the required size should be not less then value returned by the detail() method of the exception object.
> If info is zero, then the diagonal element of some of Ui is zero, and the solve could not be completed. The indexes of such matrices in the batch can be obtained with the ids() method of the exception object. You can obtain the indexes of the first zero diagonal elements in these Ui matrices using the infos() method of the exception object.

I wish we had an example demonstrating the way to handle this, but you can check how it is done in `potrf_batch_buffer.cpp` unit test: 
https://github.com/intel-innersource/libraries.performance.math.mkl/blob/develop/qa/testbase/tbmkl/tests/lapack/unit/dpcpp/potrf_batch_buffer/potrf_batch_buffer.cpp
Please note, as per docs, `batch_error` has `info()` equaling zero, to distinguish computation errors, and only `batch_error.exceptions()` will have non-zero `info()` for corresponding problem in the batch.

Reply:
This is the catch example for the batch_error:
```
    } catch(oneapi::mkl::lapack::batch_error const& be) {
        // Handle LAPACK related exceptions happened during synchronous call
        std::cout << "Unexpected exception caught during synchronous call to LAPACK API:\nReason: " << be.what() << "\nNumber: " << be.info() << std::endl;
        int i = 0;
        auto &ids = be.ids();
        for (auto const& e : be.exceptions()) {
            try {
                std::rethrow_exception(e);
            } catch(oneapi::mkl::lapack::exception &e) {
               std::cout << "Exception " << ids[i++] << " in a batch says:\n" << e.what() << " (info code: " << e.info() << ")" << std::endl;
                info = e.info();
            }
        }
    }
```

Reply:
ok, I will fix it.

Reply:
I've updated potrf_batch test. Please check if it helps:
https://github.com/intel-innersource/libraries.performance.math.mkl/blob/d962a39dc3a49c56a9ed5a1656b5fc9d01a07f41/qa/testbase/tbmkl/tests/lapack/unit/dpcpp/potrf_batch_buffer/potrf_batch_buffer.cpp